src/ngx_http_lua_api.c

```diff
@@ -194,13 +198,12 @@ ngx_http_lua_shared_memory_init(ngx_shm_zone_t *shm_zone, void *data)
 
     lmcf->shm_zones_inited++;
 
-    if (lmcf->shm_zones_inited == lmcf->shm_zones->nelts
-        && lmcf->init_handler)
+    if (lmcf->shm_zones_inited == lmcf->shm_zones->nelts)
     {
         saved_cycle = ngx_cycle;
         ngx_cycle = ctx->cycle;
 
-        rc = lmcf->init_handler(ctx->log, lmcf, lmcf->lua);
+        rc = ngx_http_lua_delay_init_phase();
```

Comment:
@agentzh It will not delayed. Because no one use `ngx_http_lua_delay_list_register` to declare need to delay `init_by_lua`, and `init_by_lua` will still run [here](https://github.com/openresty/lua-nginx-module/pull/1050/files#diff-c91b2ec425e9be9d56fee2df662b67d2R745)